app/Console/Commands/DeleteUsersCommand.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Rogue\Console\Commands;
+
+use League\Csv\Reader;
+use Rogue\Models\Post;
+use Rogue\Services\AWS;
+use Rogue\Models\Signup;
+use Rogue\Services\Fastly;
+use Illuminate\Console\Command;
+
+class DeleteUsersCommand extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'rogue:delete {input=php://stdin} {--id_column=id}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Delete signups and posts, given a CSV of user IDs.';
+
+    /**
+     * Execute the console command.
+     *
+     * @param AWS $aws
+     * @param Fastly $fastly
+     *
+     * @return mixed
+     */
+    public function handle(AWS $aws, Fastly $fastly)
+    {
+        $input = file_get_contents($this->argument('input'));
+        $csv = Reader::createFromString($input);
+        $csv->setHeaderOffset(0);
+
+        $this->line('Deleting activity for ' . count($csv) . ' users...');
+
+        foreach ($csv->getRecords() as $record) {
+            $id = $record[$this->option('id_column')];
+
+            // Find the user's signups, anonymize, and soft-delete.
+            $signups = Signup::withTrashed()->where('northstar_id', $id);
+            foreach ($signups->cursor() as $signup) {
+                $signup->update([
+                    'why_participated' => null,
+                    'details' => null,
+                ]);
+
+                $signup->delete();
+            }
+
+            // Find the user's posts, anonymize & delete images, and soft-delete:
+            $posts = Post::withTrashed()->where('northstar_id', $id);
+            foreach ($posts->cursor() as $post) {
+                $aws->deleteImage($post->url);
+                $fastly->purgeKey('post-'.$post->id);
+
+                $post->update([
+                    'text' => null,
+                    'details' => null,
+                    'location' => null,
+                    'url' => null,
+                ]);
+
+                $post->delete();
+            }
+
+            $this->info('Deleted: ' . $id . '(' . $posts->count() . ' posts and ' . $signups->count() . ' signups)');
+        }
```

Comment:
Same question as in Northstar, would it be helpful to have a final "Done anonymizing all signups and posts" at the end here so it's easy to tell when the script has finished?

Reply:
Added this in 889dbad.